### PR TITLE
2020 11 02 cleanup

### PR DESCRIPTION
**Pull request by Christewart**
Clean up a bunch of things intellij gives compiler warnings for in `core-test` when trying to execute the test suite from intellij

### CODE REVIEW

File: chain-test/src/test/scala/org/bitcoins/chain/blockchain/ChainHandlerTest.scala [hunk 1]
```diff
@@ -10,22 +10,21 @@ import org.bitcoins.core.number.{Int32, UInt32}
 import org.bitcoins.core.p2p.CompactFilterMessage
 import org.bitcoins.core.protocol.BlockStamp
 import org.bitcoins.core.protocol.blockchain.BlockHeader
-import org.bitcoins.core.util.{FutureUtil, TimeUtil}
+import org.bitcoins.core.util.{TimeUtil}
```
Comment (benthecarman) [inline chain-test/src/test/scala/org/bitcoins/chain/blockchain/ChainHandlerTest.scala]: I think having a single item with `{}` will cause a warning, should remove them, doing ctrl + alt + o, will automatically do this. Could be worth running optimize imports on the whole project if we are doing a cleanup

File: chain-test/src/test/scala/org/bitcoins/chain/blockchain/ChainHandlerTest.scala [hunk 2]
```diff
@@ -10,22 +10,21 @@ import org.bitcoins.core.number.{Int32, UInt32}
...
 import org.bitcoins.testkit.chain.{
   BlockHeaderHelper,
   ChainDbUnitTest,
   ChainTestUtil,
   ChainUnitTest
 }
-import org.bitcoins.testkit.util.{FileUtil, ScalaTestUtil}
-import org.scalatest.{Assertion, FutureOutcome}
+import org.bitcoins.testkit.util.FileUtil
+import org.scalatest.{FutureOutcome}
```
Comment (benthecarman) [inline chain-test/src/test/scala/org/bitcoins/chain/blockchain/ChainHandlerTest.scala]: ditto

File: chain-test/src/test/scala/org/bitcoins/chain/models/CompactFilterHeaderDAOTest.scala [hunk 3]
```diff
@@ -1,17 +1,12 @@
...
   ChainDbUnitTest,
-  ChainTestUtil,
-  ChainUnitTest
-}
-import org.bitcoins.testkit.core.gen.{
-  BlockchainElementsGenerator,
-  CryptoGenerators
+  ChainTestUtil
 }
+import org.bitcoins.testkit.core.gen.{CryptoGenerators}
```
Comment (benthecarman) [inline chain-test/src/test/scala/org/bitcoins/chain/models/CompactFilterHeaderDAOTest.scala]: ditto

File: project/Deps.scala [hunk 4]
```diff
@@ -203,6 +204,9 @@ object Deps {
     val scalameter =
       "com.storm-enroute" %% "scalameter" % V.scalameterV % "test" withSources () withJavadoc ()
 
+    val scalaCollectionCompat =
+      "org.scala-lang.modules" %% "scala-collection-compat" % V.scalaCollectionCompatV
```
Comment (benthecarman) [inline project/Deps.scala:208]: what is this for?

Comment (Christewart) [reply project/Deps.scala:208]: This is needed to allow `@nowarn` usage on Scala 2.12.x . The annotation was added in the scala std library in the 2.13.2 release.

**Review by benthecarman — APPROVED** [review]

**Pull request merged**